### PR TITLE
GitHub Actions workflow to validate go.mod, go.sum, and vendor directory

### PR DESCRIPTION
**Pull request by zrhoffman**
## What does this PR (Pull Request) do?

As a follow-up to #5229, this PR
- Adds a GitHub Actions workflow to verify that `go.mod`, `go.sum`, and the `vendor` directory have no uncommitted Git changes when modules are vendored.
- Removes an unused dependency and updates `go.sum`
- Gets `go mod vendor` to run successfully in the CiaB Traffic Ops `integration` service's Dockerfile by including the `v4-client` directory. Error from `go mod vendor` that is fixed:
  ```go
  #29 28.09 github.com/apache/trafficcontrol/traffic_ops/testing/api/v4 tested by
  #29 28.09       github.com/apache/trafficcontrol/traffic_ops/testing/api/v4.test imports
  #29 28.09       github.com/apache/trafficcontrol/traffic_ops/v4-client: no matching versions for query "latest"
  ```
- Gets the Dockerized TO and TM unit tests to working again by including go.mod/go.sum in the Docker image.
- [x] is not related to any Issue

## Which Traffic Control components are affected by this PR?

- CDN in a Box
- CI tests

## What is the best way to verify this PR?

* Verify the GitHub Action is working as intended.
  - Check the result of the *Check Go modules* workflow at 0e54b997ff. The following steps are expected to have failed:
    * Untracked changes to vendored dependencies
    * Changes to go.sum

  - Check the result of the *Check Go modules* workflow at fb9f772b25. It is expected to succeed.

* Run the TO API tests in CDN in a Box. On my computer, it took 55 minutes from the time when the API tests started running.
  ```shell
  docker-compose -f docker-compose.yml -f docker-compose.traffic-ops-test.yml up --no-deps -d edge dns db smtp trafficops-perl trafficops trafficvault integration;
  docker-compose -f docker-compose.traffic-ops-test.yml logs -f integration;
  ```

* Run the Traffic Monitor unit tests in Docker
  ```shell
  docker-compose -f traffic_monitor/tests/docker-compose.yml up --build
  ```

* Run the Traffic Ops unit tests in Docker
  ```shell
  docker-compose -f traffic_ops/app/bin/tests/docker-compose.yml up --build
  ```

## If this is a bug fix, what versions of Traffic Control are affected?

- master (09e25013fd)

## The following criteria are ALL met by this PR

- [x] This PR includes tests
- [x] This PR includes documentation
- [x] An update to CHANGELOG.md is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile [hunk 1]
```diff
@@ -56,9 +57,11 @@ COPY ./go.mod ./go.sum /go/src/github.com/apache/trafficcontrol/
...
 RUN go mod vendor -v
 
-RUN CGO_ENABLED=0 go test -c ./v1* -ldflags="-w -s" -o traffic_ops_v1_integration_test
-RUN CGO_ENABLED=0 go test -c ./v2* -ldflags="-w -s" -o traffic_ops_v2_integration_test
-RUN CGO_ENABLED=0 go test -c ./v3* -ldflags="-w -s" -o traffic_ops_v3_integration_test
+ENV CGO_ENABLED=0
+RUN go test -c ./v1 -ldflags="-w -s" -o traffic_ops_v1_integration_test
+RUN go test -c ./v2 -ldflags="-w -s" -o traffic_ops_v2_integration_test
+RUN go test -c ./v3 -ldflags="-w -s" -o traffic_ops_v3_integration_test
+RUN go test -c ./v4 -ldflags="-w -s" -o traffic_ops_v4_integration_test
```
Comment (ocket8888) [inline infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile:64]: Can these be a single cache layer?

File: infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile [hunk 2]
```diff
@@ -47,6 +47,7 @@ COPY ./lib/ /go/src/github.com/apache/trafficcontrol/lib/
 COPY ./traffic_ops/v1-client/ /go/src/github.com/apache/trafficcontrol/traffic_ops/v1-client/
 COPY ./traffic_ops/v2-client/ /go/src/github.com/apache/trafficcontrol/traffic_ops/v2-client/
 COPY ./traffic_ops/v3-client/ /go/src/github.com/apache/trafficcontrol/traffic_ops/v3-client/
+COPY ./traffic_ops/v4-client/ /go/src/github.com/apache/trafficcontrol/traffic_ops/v4-client/
 COPY ./traffic_ops/client/ /go/src/github.com/apache/trafficcontrol/traffic_ops/client/
```
Comment (ocket8888) [inline infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile:51]: Copying in this symlink feels extraneous

File: infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile [hunk 3]
```diff
@@ -82,6 +85,7 @@ COPY ./infrastructure/cdn-in-a-box/traffic_ops_integration_test/config.sh /opt/i
 COPY ./traffic_ops/testing/api/v1/tc-fixtures.json /opt/integration/app/tc-fixtures-v1.json
 COPY ./traffic_ops/testing/api/v2/tc-fixtures.json /opt/integration/app/tc-fixtures-v2.json
 COPY ./traffic_ops/testing/api/v3/tc-fixtures.json /opt/integration/app/tc-fixtures-v3.json
+COPY ./traffic_ops/testing/api/v4/tc-fixtures.json /opt/integration/app/tc-fixtures-v4.json
```
Comment (ocket8888) [inline infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile:88]: I'm pretty sure these four copies can be a single cache layer

File: infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile [hunk 4]
```diff
@@ -91,6 +95,9 @@ COPY --from=integration-builder \
 COPY --from=integration-builder \
     /go/src/github.com/apache/trafficcontrol/traffic_ops/testing/api/traffic_ops_v3_integration_test \
     /opt/integration/app/
+COPY --from=integration-builder \
```
Comment (ocket8888) [inline infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile:98]: Can these `COPY --from=integration-builder`s be a single cache layer?

File: infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile (hunk 3 above)
Comment (zrhoffman) [reply infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile:88]: It doesn't just copy the files, it gives each of them a new name. I'm not sure how to make that a single cache layer.

File: infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile (hunk 4 above)
Comment (zrhoffman) [reply infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile:98]: Consolidated them to a single cache layer in 6c3507ac8c.

File: infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile (hunk 2 above)
Comment (zrhoffman) [reply infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile]: No longer copying the symlink in d23aea283b.

File: infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile (hunk 1 above)
Comment (zrhoffman) [reply infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile]: Made it a single cache layer in 99c1b1af37

File: infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile (hunk 3 above)
Comment (ocket8888) [reply infrastructure/cdn-in-a-box/traffic_ops_integration_test/Dockerfile:87]: Oh, so it does.

File: infrastructure/cdn-in-a-box/enroller/enroller.go [hunk 5]
```diff
@@ -33,7 +33,7 @@ import (
 
 	log "github.com/apache/trafficcontrol/lib/go-log"
 	tc "github.com/apache/trafficcontrol/lib/go-tc"
-	"github.com/apache/trafficcontrol/traffic_ops/client"
+	"github.com/apache/trafficcontrol/traffic_ops/v4-client"
```
Comment (ocket8888) [inline infrastructure/cdn-in-a-box/enroller/enroller.go:36]: I believe this line is causing the build failure:
```
2021-02-04T18:51:49.1234730Z #16 17.29 ./enroller.go:680:29: toSession.CreateServerWithHdr undefined (type *session has no field or method CreateServerWithHdr)
```
The fix would be to link the enroller against the "stable" API client - v3 - or to update references to `CreateServerWithHdr` to simply `CreateServer` (as there is no version in v4 that doesn't use a header).

Comment (zrhoffman) [reply infrastructure/cdn-in-a-box/enroller/enroller.go:36]: Using `CreateServer()` in 97ea08c368.

**Review by ocket8888 — APPROVED** [review]
LGTM - also happy to get rid of some more references to the un-versioned client

**Pull request merged**